### PR TITLE
[ao][docs] Add description of qconfig and qengine to quantization page

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63270
* __->__ #63582

Summary:
Current quantization docs do not define qconfig and qengine. Added text to define these concepts before they are used.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30658656](https://our.internmc.facebook.com/intern/diff/D30658656)

### CODE REVIEW

File: docs/source/quantization.rst [hunk 1]
```diff
@@ -35,6 +35,13 @@ that perform all or part of the computation in lower precision. Higher-level
 APIs are provided that incorporate typical workflows of converting FP32 model
 to lower precision with minimal accuracy loss.
 
+Quantization requires users to be aware of three concepts:
+1. Quantization Config (Qconfig): Specifies how weights and activations are to be quantized. Qconfig is needed to create a quantized model.
+2. Backend: Refers to kernels that support quantization, usually with different numerics.
+3. Quantization engine (torch.backends.quantization.engine): When a quantized model is executed, the qengine specifies which
+backed is to be used for execution. It is important to ensure that the qengine is consistent with the Qconfig.
```
Comment (supriyar) [inline docs/source/quantization.rst:42]: typo: backed -> backend

Comment (supriyar) [reply docs/source/quantization.rst:42]: Is the new line here required?

File: docs/source/quantization.rst [hunk 2]
```diff
@@ -45,7 +52,8 @@ Today, PyTorch supports the following backends for running quantized operators e
 * ARM CPUs (typically found in mobile/embedded devices), via
   `qnnpack` (`<https://github.com/pytorch/QNNPACK>`_).
 
-The corresponding implementation is chosen automatically based on the PyTorch build mode.
+The corresponding implementation is chosen automatically based on the PyTorch build mode, though users
+have the option to override this by setting `torch.backends.quantization.engine` to `fbgemm` or `qnnpack`.
```
Comment (supriyar) [inline docs/source/quantization.rst:56]: maybe adding a small code-block here specifying how to override would be helpful

**Review by supriyar — APPROVED** [review]
looks good, left a few comments inline

**Comment by raghuramank100** [comment]
@raghuramank100 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30658656).

**Pull request closed without merging**